tests/test_meta_model.py

```
@@ -0,0 +1,252 @@
+    with pytest.warns(UserWarning, match="deterministic"):
+        SmabMetaModel[Beta](actions={"a1": Beta()})
+
+
+# ---------------------------------------------------------------------------
+# Type enforcement on typed subclasses (Issue #3 regression test)
+# ---------------------------------------------------------------------------
+
+
+def test_smab_meta_model_rejects_wrong_action_type():
+    """SmabMetaModel[Beta] rejects BetaMO instances (wrong concrete type)."""
+    with pytest.raises(Exception):
```

Ruff (0.15.13)

[warning] Do not assert blind exception: `Exception`

(B017)